.github/workflows/release.yml

```diff
@@ -2,47 +2,145 @@ name: Release
 
 on:
   workflow_dispatch:
+    inputs:
+      upload_to_TestPyPI:
+        description: 'upload to TestPyPI'
+        required: true
+        default: false
+        type: boolean
+
+  pull_request:
+    branches:
+      - main
+    types:
+      - closed
 
 jobs:
+
+  wheels_on_PR:
+    if: ${{ github.event.pull_request.merged == true }}
```

Comment:
Is this temporary? We do not want wheels to build on every merged PR. We want wheels to be build on releases, unless you plan to release "nightlies".

Reply:
No, this supposed to be permanent. The wheels building is tested only for Linux. There is another task `wheels` which runs on release.